src/ripple/beast/container/detail/aged_ordered_container.h

```diff
@@ -189,15 +187,13 @@ class aged_ordered_container
         : public beast::detail::empty_base_optimization<Compare>
 #ifdef _LIBCPP_VERSION
         ,
-          public std::binary_function<Key, element, bool>
```

Comment:
what would adding the double underscore do?

Reply:
This takes advantage of a libc++-only feature.  `std::binary_function` is marked deprecated.  But as an implementation technique of libc++, `std::__binary_function` has the same functionality as `std::binary_function` but is not marked deprecated.  This is done has a hack in libc++ to avoid rewriting things that used to depend on `std::binary_function` but no longer do.  And that hack is reused here.

If the hack goes away (libc++ comes into conformance), then we will get a compile-time error, and all we'll have to do is remove the code within `#ifdef _LIBCPP_VERSION`.

Reply:
thanks!